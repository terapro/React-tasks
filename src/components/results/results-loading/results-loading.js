import React from 'react';
import PropTypes from 'prop-types';
import {Spiner} from 'src/components/common/spiner/spiner';


export const ResultsLoading = ({phrase, type}) => (
  <div className='results-loading'>
    Searching <span>&quot;{phrase}&quot;</span> by <span>{type}</span>
    <Spiner/>
  </div>

);

ResultsLoading.propTypes = {
  phrase: PropTypes.string,
  type: PropTypes.string

};