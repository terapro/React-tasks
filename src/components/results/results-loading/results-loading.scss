.results-loading {
  min-height: 110px;
  width: inherit;
  text-align: center;
  font-size: 40px;
  padding-top: 200px;
}
.results-loading span {
  color: #005cbf;
}