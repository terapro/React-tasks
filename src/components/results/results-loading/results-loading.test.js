import React from 'react';
import {ResultsLoading} from "src/components/results/results-loading/results-loading";
import {shallow} from 'enzyme';


const mockTitle ='happyness';
const mockType = 'life';

describe('<ResultsLoading />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(
        <ResultsLoading phrase={mockTitle} type={mockType}/>
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});