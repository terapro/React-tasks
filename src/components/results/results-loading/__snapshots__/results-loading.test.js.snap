// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsLoading /> Rendering should render the component 1`] = `
<div
  className="results-loading"
>
  Searching 
  <span>
    "
    happyness
    "
  </span>
   by 
  <span>
    life
  </span>
  <Spiner />
</div>
`;
